modules/wakedynamics/src/WakeDynamics.f90

```diff
@@ -917,19 +1028,293 @@ subroutine updateVelocityPolar()
+      w_mean = w_mean / (iIn)
+      Vy_curl(:, :) = Vy_curl(:, :) - v_mean
+      Vz_curl(:, :) = Vz_curl(:, :) - w_mean
+   endif
+
+end subroutine
+
+
+!> This subroutine computes the near wake correction : Vx_wake  
+subroutine AddSwirl(r, Vt_wake, y, z, Vy_curl, Vz_curl)
+   real(ReKi),                   intent(in) :: r(:)       !< Tangential wake velocity deficit at first plane
+   real(ReKi),                   intent(in) :: Vt_wake(:)       !< Tangential wake velocity deficit at first plane
+   real(ReKi), dimension(:),   intent(in)      :: y                 !< Spanwise Cartesian coordinate (m)
+   real(ReKi), dimension(:),   intent(in)      :: z                 !< Wall-normal Cartesian coordinate (m)
+   real(ReKi), dimension(:,:), intent(inout)   :: Vy_curl           !< Curl velocity in the y direction (m/s)
+   real(ReKi), dimension(:,:), intent(inout)   :: Vz_curl           !< Curl velocity in the z direction (m/s)
+
+   real(ReKi) :: alpha
+   integer(IntKi) :: iz, iy, iLow, nr
+   real(ReKi) :: r_tmp, r_max
+   real(ReKi) :: Vt, S, C ! Sine and cosine
+   nr = size(r)
+   r_max = r(nr)
+   iLow=0
+   ! Loop through all plane points
+   do iz = 1,size(z)
+      do iy = 1,size(y)
+         ! Project into cartesian
+         r_tmp =  sqrt(y(iy)**2 + z(iz)**2) 
+         if (r_tmp<=r_max) then
+            if (EqualRealNos(r_tmp, 0.0_ReKi) ) then
+               S = 0.0_ReKi 
+               C = 0.0_ReKi
+            else
+               S=z(iz)/r_tmp
+               C=y(iy)/r_tmp
+            endif
+            Vt = InterpBin(r_tmp, r, Vt_wake, iLow, nr) !( XVal, XAry, YAry, ILo, AryLen )
+            Vy_curl(iy, iz) = Vy_curl(iy, iz) + Vt * S
+            Vz_curl(iy, iz) = Vz_curl(iy, iz) - Vt * C
+         endif
+      enddo
+   enddo
+   
+end subroutine AddSwirl
+
+!> Test the curled wake velocity curl function
+subroutine WD_TEST_AddVelocityCurl()
+  
+   real(ReKi) :: Vy_curl(2,2)=0.0_ReKi
+   real(ReKi) :: Vy_curl_ref(2,2)
+   real(ReKi) :: Vz_curl(2,2)=0.0_ReKi
+   real(ReKi) :: Vz_curl_ref(2,2)
+   real(ReKi) :: y(2)=(/ 0., 2./)
+   real(ReKi) :: z(2)=(/-1.,1./)
+
+   call AddVelocityCurl(Vx=10., yaw_angle=0.1, nVortex=100, R=63., psi_skew=0.2, &
+      y=y, z=z, Ct_avg=0.7, sigma_d=0.2, Vy_curl=Vy_curl, Vz_curl=Vz_curl)
+
+   if (abs(Vy_curl(1,1)-0.2171091922)>1e-5) then
+      print*,'Test fail for vy'
+      !STOP
+   endif
+   if (abs(Vz_curl(2,2)-4.45974602276e-2)>1e-5) then
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      print*,'>>>>>>>>>>>>>> Test fail for vz'
+      !STOP
+   endif
+end subroutine
+
+
+
+!> Weighted average of two angles
+subroutine filter_angles2(psi_filt, chi_filt, psi, chi, alpha, alpha_bar)
```

Comment:
This SUBROUTINE looks like exactly what I had laid out.  How do the results look?  Does theta(2) show up as nonzero?